packages/core/src/VRange.ts

```diff
@@ -233,6 +233,51 @@ export class VRange {
         }
         return this;
     }
+    /**
+     * Split the range containers up to their common ancestor. Return all
+     * children of the common ancestor that are targeted by the range after the
+     * split. If a predicate is given, splitting continues up to and including
+     * the node closest to the common ancestor that matches the predicate.
+     *
+     * @param predicate
+     */
+    split<T>(predicate?: Predicate<T>): VNode[] {
+        const ancestor = this.startContainer.commonAncestor(this.endContainer);
+        const closest = ancestor.closest(predicate);
+        const container = closest ? closest.parent : ancestor;
+
+        // Split the start ancestors.
+        let start = this.start;
+        do {
+            let startAncestor = start.parent;
+            if (startAncestor.children.indexOf(start) > 0) {
```

Comment:
Comment would be nice:
`// Don't split at the start edge of a node.`
But shouldn't you also check that we're not at the _end_ edge?